src/overlays/actors/ovl_Boss_Dodongo/z_boss_dodongo.c

```diff
@@ -51,311 +44,1718 @@ const ActorInit Boss_Dodongo_InitVars = {
+            temp_s3[i + temp2] = sp54[i + i2];
+        }
+    }
+}
+
+void func_808C17C8(GlobalContext* globalCtx, Vec3f* arg1, Vec3f* arg2, Vec3f* arg3, f32 arg4, s16 arg5) {
+    s16 i;
+    BossDodongoEffect* eff = (BossDodongoEffect*)globalCtx->specialEffects;
+
+    for (i = 0; i < arg5; i++, eff++) {
+        if (eff->unk_24 == 0) {
+            eff->unk_24 = 1;
+            eff->unk_0 = *arg1;
+            eff->unk_C = *arg2;
+            eff->unk_18 = *arg3;
+            eff->unk_2C = arg4 / 1000.0f;
+            eff->alpha = 255;
+            eff->unk_25 = (s16)Rand_ZeroFloat(10.0f);
+            break;
+        }
+    }
+}
+
+s32 BossDodongo_AteExplosive(BossDodongo* this, GlobalContext* globalCtx) {
+    f32 dx;
+    f32 dy;
+    f32 dz;
+    Actor* currentExplosive = globalCtx->actorCtx.actorLists[ACTORCAT_EXPLOSIVE].head;
+    Actor* thisx = &this->actor;
+
+    while (currentExplosive != NULL) {
+        if (currentExplosive == thisx) {
+            currentExplosive = currentExplosive->next;
+            continue;
+        }
+
+        dx = currentExplosive->world.pos.x - this->mouthPos.x;
+        dy = currentExplosive->world.pos.y - this->mouthPos.y;
+        dz = currentExplosive->world.pos.z - this->mouthPos.z;
+
+        if ((fabsf(dx) < 40.0f) && (fabsf(dy) < 40.0f) && (fabsf(dz) < 40.0f)) {
+            Actor_Kill(currentExplosive);
+            return true;
+        }
+
+        currentExplosive = currentExplosive->next;
+    }
+
+    return false;
+}
+
+void BossDodongo_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BossDodongo* this = THIS;
+    s16 i;
+    u16* temp_s1_3;
+    u16* temp_s2;
+    u32 temp_v0;
+
+    globalCtx->specialEffects = &this->effects;
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 9200.0f, ActorShadow_DrawCircle, 250.0f);
+    Actor_SetScale(&this->actor, 0.01f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_0601B310, &D_0600F0D8, NULL, NULL, 0);
+    Animation_PlayLoop(&this->skelAnime, &D_0600F0D8);
+    this->unk_1F8 = 1.0f;
+    BossDodongo_SetupIntroCutscene(this, globalCtx);
+    this->health = 12;
+    this->fogMin = 995.0f;
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->fogMax = 1000.0f;
+    this->unk_224 = 2.0f;
+    this->unk_228 = 9200.0f;
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit, this->items);
+
+    if (Flags_GetClear(globalCtx, globalCtx->roomCtx.curRoom.num)) { // KD is dead
+        temp_s1_3 = SEGMENTED_TO_VIRTUAL(&D_030021D8);
+        temp_s2 = SEGMENTED_TO_VIRTUAL(gBossDodongoLavaFloorRockTex);
+
+        Actor_Kill(&this->actor);
+        Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, -890.0f, -1523.76f,
+                           -3304.0f, 0, 0, 0, 0);
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_BREAKWALL, -890.0f, -1523.76f, -3304.0f, 0, 0, 0, 0x6000);
```

Comment:
breakwall type enum. theres a couple in this file that you need to replace

Reply:
Not really sure how I should use it since the enum does 0-3 and this is 0x6000

Reply:
ah yea this has a few things packed in it, which we dont have macros for. nvm